### DISCUSSION TITLE
Feature Request: Hex/binary/octal numbers?

### DISCUSSION THREAD

**Discussion by rongcuid** [duckdb/duckdb#5700] [Ideas]
It would be nice if the SQL supports `0x`, `0b`, `0`, prefixes for supporting other integer radix.

**Comment by manticore-projects** [comment]
At least there could be some documentation on how to deal with Hex numbers. Right now there is nothing?
From what I have found by trial'n error:

```sql

-- Unsupported Hex Numbers, expected result: 1
SELECT BIT_AND(x) as bit_and FROM UNNEST([0xF001, 0x00A1]) as x;

-- can be rewritten to

SELECT Bit_Count(BIT_AND(x)) as bit_and 
FROM (SELECT UNNEST([ From_Hex('F001')::BIT, From_Hex('00A1')::BIT]) as x);

/* returns
┌─────────┐
│ bit_and │
├─────────┤
│       1 │
└─────────┘
*/

**Comment by manticore-projects** [comment]
As a work around, I just have pushed https://github.com/JSQLParser/JSqlParser/commit/620db709e48c22e049c449d1d4e1d9b0fb624584 to translate Hex into Longs.
One can overwrite the `HexValue` Visitor to write out `LongValue` instead:
```java
public void visit(HexValue hexValue) {
    super.visit( hexValue.getLongValue() );
  }
```

This allows for rewriting the statement above into: 

``` sql
-- replace 0xF001 or X'00A1'
SELECT BIT_AND(x) as bit_and
FROM (SELECT UNNEST([61441, 161]) as x) as x;

/* result
┌─────────┐
│ bit_and │
├─────────┤
│       1 │
└─────────┘
*/

```
